### PR TITLE
Handle Buffer obfuscation

### PR DESCRIPTION
**Pull request by pedrobranco**
This PR adds support for correct obfuscation of Buffers.

### CODE REVIEW

File: test/src/index.test.js [hunk 1]
```diff
@@ -37,6 +37,18 @@ describe('Anonymizer', () => {
       });
     });
 
+    it(`should obfuscate keys which type is Buffer`, () => {
```
Comment (waldyrious) [inline test/src/index.test.js]:
```suggestion
    it(`should obfuscate keys whose type is Buffer`, () => {
```

File: test/src/index.test.js [hunk 2]
```diff
@@ -37,6 +37,18 @@ describe('Anonymizer', () => {
...
+      const anonymize = anonymizer();
+
+      expect(anonymize({ foo: Buffer.from('foobarfoobar') })).toEqual({ foo: '--REDACTED--' });
+    });
+
+    it(`should not obfuscate keys which type is Buffer and are whitelisted`, () => {
```
Comment (waldyrious) [inline test/src/index.test.js]: Maybe clearer this way:

```suggestion
    it(`should not obfuscate Buffer-type keys that are whitelisted`, () => {
```

**Comment by pedrobranco** [comment]
@waldyrious Did you test this PR? Can you merge it?

**Comment by waldyrious** [comment]
I didn't test it, but I can give it a spin and merge. Will do so shortly.

**Comment by pedrobranco** [comment]
> I didn't test it, but I can give it a spin and merge. Will do so shortly.

Thanks!

**Comment by waldyrious** [comment]
Works as expected:

- On the master branch:
  ```node
  > const anonymizer = require('src');
  > const anonymize = anonymizer(['foo']);
  > anonymize({ foo: Buffer.from('asdf') })
  {
    foo: {
      type: '--REDACTED--',
      data: [ '--REDACTED--', '--REDACTED--', '--REDACTED--', '--REDACTED--' ]
    }
  }
  > anonymize({ bar: Buffer.from('asdf') })
  {
    bar: {
      type: '--REDACTED--',
      data: [ '--REDACTED--', '--REDACTED--', '--REDACTED--', '--REDACTED--' ]
    }
  }
  ```
- On this PR's branch:
  ```node
  > const anonymizer = require('src');
  > const anonymize = anonymizer(['foo']);
  > anonymize({ foo: Buffer.from('asdf') })
  { foo: <Buffer 61 73 64 66> }
  > anonymize({ bar: Buffer.from('asdf') })
  { bar: '--REDACTED--' }
  ```

**Pull request merged**